### PR TITLE
add a command to report message

### PR DESCRIPTION
**Pull request by dtinth**
# Description

Per discussion with admins, they would like a command to easily report a message. When clicked, the message URL will be sent to the moderator channel.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or other

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by masuru-aomsin — APPROVED** [review]

**Comment by SaltyAom** [comment]
kk, will review

File: src/types/CommandHandlerExecutor.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { CommandInteraction } from 'discord.js'
+
+import type Bot from '../client.js'
```
Comment (SaltyAom) [inline src/types/CommandHandlerExecutor.ts:3]: It's `client.ts` instead of `client.js`

File: src/types/EventHandlerExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { ClientEvents } from 'discord.js'
+
+import type Bot from '../client.js'
```
Comment (SaltyAom) [inline src/types/EventHandlerExecutor.ts:3]: resolve this to `.ts` instead of `.js`

**Review by SaltyAom — CHANGES_REQUESTED** [review]
Please resolve comments file extension to `.ts`

**Pull request merged**